dev-libs/libbase58/libbase58-0.1.4.ebuild

```diff
@@ -11,18 +11,37 @@ LICENSE="MIT"
 SRC_URI="https://github.com/luke-jr/${PN}/archive/v${PV}.tar.gz -> ${P}.tar.gz"
 SLOT="0/0"
 KEYWORDS="amd64 ~arm ~mips ~ppc ~ppc64 x86"
-IUSE="tools"
+IUSE="test tools"
+RESTRICT="!test? ( test )"
 
-DEPEND=""
-RDEPEND="${DEPEND}"
+# NOTE: If not testing, we don't need non-native libgcrypt
+RDEPEND="tools? ( dev-libs/libgcrypt )"
+DEPEND="${RDEPEND}
+	test? ( dev-libs/libgcrypt[${MULTILIB_USEDEP}] )
+"
+
+ECONF_SOURCE="${S}"
 
 src_prepare() {
 	eapply_user
 	eautoreconf
 }
 
-src_configure() {
-	econf \
-		--disable-static \
-		$(use_enable tools tool)
+multilib_src_configure() {
+	local myeconf=(
+		--disable-static
+		LIBGCRYPT_CONFIG="${EPREFIX}/usr/bin/${CHOST}-libgcrypt-config"
+	)
+	if multilib_is_native_abi && use tools; then
+		myeconf+=( --enable-tool )
+	elif use test; then
+		myeconf+=( --enable-tool --bindir='/TRASH' )
+	else
+		myeconf+=( --disable-tool )
+	fi
+	econf "${myeconf[@]}"
+}
+
+multilib_src_install_all() {
+	rm -rf "${ED}/TRASH" || die
```

Comment:
We rarely use `-f` in rm, you should know when this dir exists and when not.

Reply:
Overall this seems a bit weird. What if I want to keep the tools even though enabling `test` IUSE?

Reply:
Then you should enable the `tools` USE flag, which will put the native tools in `bin` instead of `TRASH`

To make `rm` conditional would be something like (pseudo-code because I don't know how to do it): if iuse test && more than 1 ABI is enabled (or I guess we can just mkdir TRASH earlier, but what's the point?)

Reply:
```
if multilib_is_native_abi; then
```
should work.

Reply:
Not inside `multilib_src_install_all`?